src/navigation/dispatch/components/BulkEditTasksFloatingButton.js

```diff
@@ -1,50 +1,80 @@
+      addOrder: (task, taskListId) => addItem(task, taskListId, true),
+      addTask: (task, taskListId) => addItem(task, taskListId, false),
+      removeOrder: (task, taskListId) => removeItem(task, taskListId, true),
+      removeTask: (task, taskListId) => removeItem(task, taskListId, false),
+    }
+  ), [addItem, removeItem]);
 
-        if (!newSelectedTasks[taskListId]) {
-          newSelectedTasks[taskListId] = [];
-        }
+  const addItem = useCallback((task, taskListId, isAssigningOrder) => {
+    removeItem(task, taskListId, !isAssigningOrder);
 
-        if (newSelectedTasks[taskListId].some(i => i.id === task.id)) {
-          return;
-        }
+    setSelectedTasks(prevSelectedTasks => {
+      const { orders, tasks } = prevSelectedTasks;
+      const itemsList = isAssigningOrder ? orders : tasks;
+      const newItemsList = {...itemsList};
 
-        newSelectedTasks[taskListId].push(task);
-        setSelectedTasks(newSelectedTasks);
-      },
-      removeItem: (task, taskListId) => {
-        const newSelectedTasks = {...selectedTasks};
-        const selectedTasksForTaskList = newSelectedTasks[taskListId];
+      if (!newItemsList[taskListId]) {
+        newItemsList[taskListId] = [];
+      }
 
-        if(!selectedTasksForTaskList) {
-          return;
-        }
+      newItemsList[taskListId].push(task);
 
-        newSelectedTasks[taskListId] = selectedTasksForTaskList.filter(t => t.id !== task.id);
+      return {
+        orders: isAssigningOrder ? newItemsList : orders,
+        tasks: !isAssigningOrder ? newItemsList : tasks,
+      };
+    });
+  }, [removeItem]);
 
-        if (newSelectedTasks[taskListId].length === 0) {
-          delete newSelectedTasks[taskListId];
-        }
+  const removeItem = useCallback((task, taskListId, isUnassigningOrder) => {
+    setSelectedTasks(prevSelectedTasks => {
+      const { orders, tasks } = prevSelectedTasks;
+      const itemsList = isUnassigningOrder ? orders : tasks;
+      const newItemsList = {...itemsList};
+      const selectedTasksForTaskList = newItemsList[taskListId];
 
-        setSelectedTasks(newSelectedTasks);
+      if(!selectedTasksForTaskList) {
+        return prevSelectedTasks;
       }
-    }
-  }, [selectedTasks]);
+
+      newItemsList[taskListId] = selectedTasksForTaskList.filter(t => t.id !== task.id);
+
+      if (newItemsList[taskListId].length === 0) {
+        delete newItemsList[taskListId];
+      }
+
+      return {
+        orders: isUnassigningOrder ? newItemsList : orders,
+        tasks: !isUnassigningOrder ? newItemsList : tasks,
+      };
+    });
+  }, []);
 
   const allSelectedTasks = useMemo(() => {
-    return _.flatMap(Object.values(selectedTasks));
+    const ordersByTaskList = selectedTasks.orders || {};
+    const tasksByTaskList = selectedTasks.tasks || {};
+
+    const orders = _.flatMap(Object.values(ordersByTaskList));
+    const tasks = _.flatMap(Object.values(tasksByTaskList));
+
+    return [...orders, ...tasks];
```

Comment:
I'm not sure if `selectedTasks.orders` already has all the order's `taskIds`, but we need to make sure we only have all the taskIds from the order/s, merge with `tasks` and remove possible duplicates!